### DISCUSSION TITLE
Unable to scroll to bottom of the grid, getting error

### DISCUSSION THREAD

**Discussion by Pradhakrishna284** [cypress-io/cypress#23862] [Questions and Help]
Grid Name:- wj-flex-grid.bento-flex-grid
Grid Row:- div[role=\'row\']
Grid Cell:- div[role=\'gridcell\']

At a time, in the grid, only 12 rows are visible (1 header row, 1 empty row to enter data, and then the remaining 10 rows are data if there are any records), 
my grid has more than 10 data rows, my expected text("Test_TT_16") is at 18th data row, though I used
scrollIntoView() method, I am unable to move to the grid to bottom to find my text. Any help to modify the code.

		
Approach1:-	
Issue1:- cy.scrollTo() can only be used to scroll 1 element, you tried to scroll 12 elements. 
I get this issue, when I use the below code.
cy.get('wj-flex-grid.bento-flex-grid div[wj-part="root"]').scrollIntoView().within(() => {

			cy.get('div[role=\'row\']').each((row, index) => {
				cy.log('Index within the loop: -' + index)
				cy.wrap(row).within(() => { //within the selected row
	
					//1st row is header row, 2nd row is empty row to enter data in the grid, the actual data starts from 3rd row in the grid
					if (index > 1) {
						cy.get('div[role=\'gridcell\']:nth-child(1)').invoke('text').then((name) => {
							cy.log("First column value:- " + name)
							if (name == "Test_TT_16") {
								cy.log("Expected text present:- " + name)
								return false
							}
				
						})
					}

				})				
			})

			cy.get('div[role=\'row\']').scrollTo('bottom')

		})

Approach2:-
Issue2:- When I use the below code, only the first 10 visible data rows are printed, but my data is not printed, but the grid is scrolled to bottom
after printing the first 10 rows data
cy.get('wj-flex-grid.bento-flex-grid div[wj-part="root"] div[role=\'row\']', { timeout: 30000 }).each((row, index) => {
			cy.wrap(row).scrollIntoView().within(() => { //within the selected row

				if (index > 1) {
					cy.get('div[role=\'gridcell\']:nth-child(1)').invoke('text').then((name) => {
						cy.log("First column value:- " + name)
						if (name == "Test_TT_16") {
							cy.log("Expected text present:- " + name)
							return false
						}
				
					})
				}

			})	
			cy.get('wj-flex-grid.bento-flex-grid div[wj-part="root"]').scrollTo('bottom')		
				
		})

**Comment by jjhelguero** [comment]
It's a bit unclear what you are trying to do. You state `my expected text("Test_TT_16") is at 18th data row` but don't mention whether you are validating it shows up on the table or invoking an action on it.

**Reply by Pradhakrishna284** [reply]
since only the first 10 data rows are **visible** to the user, my text ("Test_TT_16") is at 18th row, validation is not also happening because the grid has to be scroll down automatically through code for the next rows to be visible to the user. But when my text appears in the first 10 rows, validation is happening.

For example, the grid has 20 data rows as shown below, at a time only the first 10 data rows are visible to the user, if the user wants to see row18, then he has to scroll down. To explain my issue with this case, at the moment the validation check happens with the first 10 data rows only because cypress can see only the first 10 data rows. It runs the test case and passes, but it is not scroll to the bottom of the grid to see all the data rows, and this is the part I am unable to implement correctly in the code.

Header Row
Empty Row to enter new data
Row1
Row2
Row3
Row4
Row5
Row6
Row7
Row8
Row9
Row10
Row11
Row12
Row13
Row14
Row15
Row16
Row17
Row18
Row19
Row20

**Comment by samelawrence** [comment]
This is a duplicate of [this other discussion here](https://github.com/cypress-io/cypress/discussions/23861). @Pradhakrishna284 please close the other thread.
